docs/mvvm/MigratingFromMvvmLight.md

```diff
@@ -0,0 +1,791 @@
+
+The first change here will be swapping using directives in your components.
+
+```csharp
+// MvvmLight
+using GalaSoft.MvvmLight.Command;
+using Galasoft.MvvmLight.CommandWpf;
+
+// Toolkit.Mvvm
+using Microsoft.Toolkit.Mvvm.Input;
+```
+
+**Note** on `RelayCommand<T>` constructors. MvvmLight uses weak references to establish the link between the command and the action called from the associated class. This is not required by the MVVM Toolkit implementation and if this optional parameter has been set to `true` in any of your constructors, this will be removed.
+
+### RelayCommand<T> Methods
+
+#### RaiseCanExecuteChanged ()
+
+The functionality of `RaiseCanExecuteChanged()` can be achieved with the MVVM Toolkit's `NotifyCanExecuteChanged()` method. 
+
+```csharp
+// MvvmLight
+var command = new RelayCommand<string>(this.OnCommand);
+command.RaiseCanExecuteChanged();
+
+// Toolkit.Mvvm
+var command = new RelayCommand<string>(this.OnCommand);
+command.NotifyCanExecuteChanged();
+```
+
+## Migrating SimpleIoc
+
+The [IoC](Ioc) implementation in the MVVM Toolkit takes advantage of existing .NET APIs through the `Microsoft.Extensions.DependencyInjection` library. 
+
+This is the biggest change between MvvmLight and the MVVM Toolkit. 
+
+This implementation will feel familiar if you've implemented dependency injection with ASP.NET Core applications.
+
+### Registering your dependencies
+
+With MvvmLight, you may have registered your dependencies similar to these scenarios using `SimpleIoc`.
+
+```csharp
+public void RegisterServices()
+{
+  SimpleIoc.Default.Register<INavigationService, NavigationService>();
+
+  SimpleIoc.Default.Register<IDialogService>(() => new DialogService());
+}
+```
+
+With the MVVM Toolkit, you would achieve the same as follows.
+
+```csharp
+public void RegisterServices()
+{
+  Ioc.Default.ConfigureServices(services => 
+  {
+    services.AddSingleton<INavigationService, NavigationService>();
+
+    services.AddSingleton<IDialogService>(new DialogService());
+  });
+}
+```
+
+### Resolving dependencies
+
+Ioc in the MVVM Toolkit, like many other service providers, is capable of constructor injection. 
+
+There are also often times in your application where you need to access the services from the service itself.
+
+In MvvmLight, you might access a service directly as follows:
+
+```csharp
+IDialogService dialogService = SimpleIoc.Default.GetInstance<IDialogService>();
+```
+
+Migrating to the MVVM Toolkit, you will achieve the same with:
+
+```csharp
+IDialogService dialogService = Ioc.Default.GetService<IDialogService>();
+```
+
+### Removing dependencies
+
+With `SimpleIoc`, you would unregister your dependencies with the following method call.
+
+```csharp
+SimpleIoc.Default.Unregister<INavigationService>();
+```
+
+There is no direct replacement for removing dependencies with the MVVM Toolkit `Ioc` implementation.
```

Comment:
We might want to mention that this difference is because service providers are meant to be immutable once initialized, and this change is a direct result of us using the `Microsoft.Extensions.DependencyInjection` library, which follows this approach as well. If developers need to alter the collection of available services while the application is running, they should build another `IServiceProvider` instance and use that instead.

In that case, they can do that by using the `MS.EX.DI` APIs directly, like so:
```csharp
IServiceProvider services = new ServiceCollection()
    .AddSingleton<INavigationService, NavigationService>()
    .AddSingleton<IDialogService>(new DialogService())
    // Other services...
    .BuildServiceProvider();
```
And then they can just use the `services` instance to call `GetResolve<TService>()` as usual 👍

Reply:
I've not bothered doing anything with this just yet. I'm still unsure what it is we are doing around the Ioc implementation. Need to catch up on convos on that one.